sdk/core/azure-core/src/main/java/com/azure/core/http/rest/PagedFlux.java

```diff
@@ -50,12 +51,26 @@ public PagedFlux(Supplier<Mono<PagedResponse<T>>> firstPageRetriever) {
         this(firstPageRetriever, token -> Mono.empty());
     }
 
+    /**
+     * Creates an instance of {@link PagedFlux} that consists of only a single page with a given element count.
+     *
+     * <p><strong>Code sample</strong></p>
+     *
+     * {@codesnippet com.azure.core.http.rest.PagedFlux.singlepage.instantiationWithPageSize}
+     *
+     * @param firstPageRetriever Function that retrieves the first page.
+     */
+    public PagedFlux(Function<Integer, Mono<PagedResponse<T>>> firstPageRetriever) {
+        this(firstPageRetriever, (token, pageSize) -> Mono.empty());
+    }
```

Comment:
This overload is used only when there is a single page to retrieve. So, why is the pageSize required?

Reply:
There are single paged response API, such as [Azure Search Autocomplete](https://docs.microsoft.com/en-us/rest/api/searchservice/autocomplete) and [Azure Search Suggestions](https://docs.microsoft.com/en-us/rest/api/searchservice/suggestions), which have the option of page size, or response element count. This is to enable support of those if `byPage(int preferredPageSize)` is used in the `PagedFlux`.